joblib/format_stack.py

```diff
@@ -196,6 +196,10 @@ def format_records(records):   # , print_globals=False):
             # keep the original file string.
             pass
         link = file
+
+        if file.endswith('.pyc'):
+            file = file[:-4] + '.py'
```

Comment:
Can you put these two lines above the `link = file` line?

This way, the filename that appears in the traceback is a .py and not a .pyc.

Bonus points if you squash your commits.